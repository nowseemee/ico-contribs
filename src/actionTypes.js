export const FETCH_CONTRIBUTORS_REQUEST = "FETCH_CONTRIBUTORS_REQUEST";
export const FETCH_CONTRIBUTORS_SUCCESS = "FETCH_CONTRIBUTORS_SUCCESS";
export const FETCH_CONTRIBUTORS_FAILURE = "FETCH_CONTRIBUTORS_FAILURE";
