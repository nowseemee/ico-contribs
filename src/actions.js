import {
  FETCH_CONTRIBUTORS_REQUEST,
  FETCH_CONTRIBUTORS_SUCCESS,
  FETCH_CONTRIBUTORS_FAILURE
} from "./actionTypes";

export const fetchContributorsRequest = () => ({
  type: FETCH_CONTRIBUTORS_REQUEST
});

export const fetchContributorsSuccess = body => ({
  type: FETCH_CONTRIBUTORS_SUCCESS,
  body
});

export const fetchContributorsFailure = error => ({
  type: FETCH_CONTRIBUTORS_FAILURE,
  error
});

export const fetchContributors = () => dispatch => {
  dispatch(fetchContributorsRequest());
  return fetch("/api/inputs")
    .then(res => res.json())
    .then(body => dispatch(fetchContributorsSuccess(body)))
    .catch(error => dispatch(fetchContributorsFailure(error)));
};
