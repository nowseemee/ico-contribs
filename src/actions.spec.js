import configureMockStore from "redux-mock-store";
import fetchMock from "fetch-mock";
import thunk from "redux-thunk";

import * as actions from "./actions";
import * as types from "./actionTypes";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe("actions", () => {
  describe("sync", () => {
    test("fetchContributorsRequest", () => {
      expect(actions.fetchContributorsRequest()).toMatchSnapshot();
    });

    test("fetchContributorsSuccess", () => {
      expect(
        actions.fetchContributorsSuccess({ preIco: [], ico: [] })
      ).toMatchSnapshot();
    });

    test("fetchContributorsFailure", () => {
      expect(
        actions.fetchContributorsFailure(new Error("wtf?"))
      ).toMatchSnapshot();
    });
  });

  describe("async", () => {
    afterEach(() => {
      fetchMock.reset();
      fetchMock.restore();
    });
    describe("fetchContributors", () => {
      test("creates FETCH_CONTRIBUTORS_SUCCESS when fetching has been done", () => {
        const body = { happy: "true" };
        fetchMock.getOnce("/api/inputs", {
          body,
          headers: { "content-type": "application/json" }
        });
        const expectedActions = [
          { type: types.FETCH_CONTRIBUTORS_REQUEST },
          {
            type: types.FETCH_CONTRIBUTORS_SUCCESS,
            body
          }
        ];
        const store = mockStore();
        return store.dispatch(actions.fetchContributors()).then(() => {
          expect(store.getActions()).toEqual(expectedActions);
        });
      });
    });
  });
});
