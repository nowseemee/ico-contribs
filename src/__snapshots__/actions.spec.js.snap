// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions sync fetchContributorsFailure 1`] = `
Object {
  "error": [Error: wtf?],
  "type": "FETCH_CONTRIBUTORS_FAILURE",
}
`;

exports[`actions sync fetchContributorsRequest 1`] = `
Object {
  "type": "FETCH_CONTRIBUTORS_REQUEST",
}
`;

exports[`actions sync fetchContributorsSuccess 1`] = `
Object {
  "body": Object {
    "ico": Array [],
    "preIco": Array [],
  },
  "type": "FETCH_CONTRIBUTORS_SUCCESS",
}
`;
